napari/utils/event.py

```diff
@@ -537,16 +539,23 @@ def _prepare_event(self, *args, **kwargs):
             event = args[0]
             # Ensure that the given event matches what we want to emit
             assert isinstance(event, self.event_class)
-        elif not args:
-            args = self.default_args.copy()
-            args.update(kwargs)
-            event = self.event_class(**args)
-        else:
-            raise ValueError(
-                "Event emitters can be called with an Event "
-                "instance or with keyword arguments only."
-            )
-        return event
+            return event
+
+        if len(args) == 1:
+            if not kwargs:
+                # if an EventEmitter is called with a single argument (that is
+                # not an instance of self.event_class), assume that that
+                # argument is a value to be emitted.
+                kwargs['value'] = args[0]
+            else:
+                raise ValueError(
+                    "Event emitters may be called with an Event instance, "
+                    "a single argument, or with keyword arguments only."
+                    f"\ngot: args={args}, kwargs={kwargs} "
+                )
+        args = self.default_args.copy()
+        args.update(kwargs)
+        return self.event_class(**args)
```

Comment:
Oof. This whole change makes me dislike the `value` kwarg even more. Perhaps the keyword can be the event name?

Reply:
I actually don't mind `value` on the event as the one place where the payload is being carried. To me this makes a lot of sense and is fairly readable, but I see your comment about `value` inside the handlers and think we should change it there. @tlambert03 thoughts?

Reply:
Sorry I’m on mobile at the moment and can’t see the full conversation.  I don’t care what you call the argument in the handler... but I agree with @sofroniewn that the event payload should be standardized.  I hated having to go find the event emitter to figure out the name of the event object attribute I’m supposed to be querying to retrieve the value.  I also don’t mind this code block either if it lets us make connections easier. 
@jni  is your main objection the argument name in the handler?  Or do you want the actual event object to have variable attribute names